File: 例子/Java的例子/pure/EncryptedDataDecrypt.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;

import net.netca.pki.encoding.asn1.pki.NetcaHasher;
import net.netca.pki.encoding.asn1.pki.NetcaSymEncrypter;
import net.netca.pki.encoding.asn1.pki.Pkcs5PBKDF2;
import net.netca.pki.encoding.asn1.pki.cms.EncryptedData;

public class EncryptedDataDecrypt {

	public static void main(String[] args) throws Exception{
		if(args.length!=3) {
			usage();
            return;
        }
		
		String pwd=args[0];
		String inFileName=args[1];
		String outFileName=args[2];
		
		byte[] cipher=readFile(inFileName);
		EncryptedData encryptedData=new EncryptedData(cipher);
		byte[] clear=encryptedData.pkcs5Decrypt(pwd, new NetcaSymEncrypter(),
				new Pkcs5PBKDF2(new NetcaHasher()));
		writeFile(outFileName,clear);
		System.out.println("Success");
		
	}
	private static void usage() {
		System.out.println("Usage: java EncryptedDataDecrypt pwd cipherFile clearFile");
	}
	
	private static byte[] readFile(String fileName) throws Exception{
		File file=new File(fileName);
		
		long fileLen=file.length();
		if(fileLen>=Integer.MAX_VALUE) {
			throw new Exception("file too big");
		}
		int len=(int)fileLen;
		byte[] data=new byte[len];
		FileInputStream fIn=new FileInputStream(file);
		
		try{
			if (fIn.read(data)!=len) {
				throw new Exception("read file fail");
			}
			return data;
		} finally {
			fIn.close();
		}
	}
	
	private static void writeFile(String fileName,byte[] data) throws Exception{
		FileOutputStream fOut=new FileOutputStream(fileName);
		try {
			fOut.write(data);
		} finally {
			fOut.close();
		}
		
	}
}

File: 例子/lua的例子/EncryptedDataDecrypt.lua
local CryptCore=require("CryptCore")

function ToANSIString(utf8String)
    return CryptCore.UTF8ToACP(utf8String)
end



function Usage(filename)
    print(string.format("Usage: lua %s pwd inFileName outFileName",filename))
    print(ToANSIString("参数"))
	print(ToANSIString("\tpwd：口令"))
	print(ToANSIString("\tinFileName：密文文件名"))
	print(ToANSIString("\toutFileName：明文文件名"))
	print(ToANSIString("说明"))
    print(ToANSIString("\t使用pwd解密inFileName的EncrypedData格式的数据，结果保存在outFileName中"))
end

function GetFileConetent(fileName)
    local fIn=io.open(fileName,"rb")
    if fIn==nil then
        return nil
    end
    local data=fIn:read("*a")
    fIn:close()
    
    return data
end

function save(outFileName,data)
	local fOut=assert(io.open(outFileName,"wb"))
	fOut:write(data)
	fOut:close()
end

if #arg~=3 then
    Usage(arg[0])
    os.exit(-1)
end

local pwd=arg[1]
local inFileName=arg[2]
local outFileName=arg[3]

local encrypedData=GetFileConetent(inFileName)
if encrypedData==nil then
    print(ToANSIString("读密文文件失败"));
    os.exit(-1)
end

local clear=CryptCore.EncryptedDataDecrypt(pwd,encrypedData)
if clear==nil then
	print(ToANSIString("解密失败"));
    os.exit(-1)
end
save(outFileName,clear)
print(ToANSIString("解密成功"));


